src/Simulation/DecompositionsCore/Type2/Type2-Decompositions.qs

```diff
@@ -0,0 +1,509 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+    open Microsoft.Quantum.Decompositions.Utilities as Utils;
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// Performs the identity operation (no-op) on a single qubit.
+    ///
+    /// # Remarks
+    /// This is a no-op. It is provided for completeness and because
+    /// sometimes it is useful to call the identity in an algorithm or to pass it as a parameter.
+    @EnableTestingViaName("Test.Decompositions.I")
+    operation I(target : Qubit) : Unit
```

Comment:
Yup, you nailed it. Again, documentation will help here, but there it's easy to have confusion between "intrinsic" meaning an operation or function that has `body intrinsic` in its implementation, "intrinsic" meaning a member of the Microsoft.Quantum.Intrinsic namespace, "intrinsic" meaning natively supported by the target simulator/hardware, and "intrinsic" meaning part of the default set of body-intrinsic operations in the Q# core. I'll clarify this in the docs, but for reference here, this is how I'm thinking about terminology:

- Quantum Gate Set: the set of gates natively supported by the underlying simulator/hardware.
- Body-Intrinsic: any operation or function whose Q# implementation is `body-intrinsic`. When an operation in the Microsoft.Quantum.Intrinsic namespace is also body-intrinsic it is often an indication that it represents a gate from the quantum gate set, but not always, as some helper methods that rely on C# implementation are body-intrinsic, but that should be rare in a decomposition package.
- Decomposition: any operation representing a quantum gate or operation that is not natively supported by the target simulator/hardware and so must be implemented by a combination of calls to helpers and body-intrinsic operations. From a circuit perspective, the decomposition is often the circuit comprised of supported gates needed to implement an unsupported gate.
- Default Intrinsics: the set of body-intrinsic operations defined in the Qsharp.Core that are part of the Microsoft.Quantum.Intrinsic namespace. Right now this is equivalent to the QuantumSimulator's quantum gate set, but that may change as simulator implementation grows and changes.

Reply:
Thank you for the explanation! In general, I'd very much prefer using different terms for the different kind of `intrinsics` than relying on the documentation to describe the different meaning  depending on the context. I realize that some of the terminology has been around for a bit and it would be hard to change it, but targeting is relatively recent -- maybe we could introduce a new term here?

Reply:
Hmm... I do agree that would be nicer, but two major limiting factor are that I need to keep the namespace the same and I can't reasonably change the `body intrinsic` keywords. So the language sort of has one of those confusions built-in. I'm not going to give up though... I firmly believe that having good terminology is crucial to being able to discuss engineering problems and solutions; if we agree how to communicate the ideas then we can't really collaborate on them well. Definitely let me know if you have any ideas for alternate, more specific terminology!